packages/babel-plugin-redux-saga/src/index.js

```diff
@@ -5,38 +5,58 @@ var globalSymbolName = '@@redux-saga/LOCATION'
 
 function getSourceCode (path){
   // use `toString` for babel v7, `getSource` for older versions
-  return Object.prototype.hasOwnProperty.call(path, 'toString') ? path.toString() : path.getSource();
+  const rawCode = Object.prototype.hasOwnProperty.call(path, 'toString') ? path.toString() : path.getSource();
+  return rawCode.replace(/^(yield\*?)\s+/, '')
 }
 
 function isSaga (path){
   return path.node.generator;
 }
 
 module.exports = function(babel) {
-  var { types: t, template } = babel
+  var { types: t } = babel
   var sourceMap = null
   var alreadyVisited = new WeakSet();
 
-  const extendExpressionWithLocation = template(`
-    (function reduxSagaSource() {
-      return Object.defineProperty(TARGET, SYMBOL_NAME, {
-        value: {
-          fileName: FILENAME,
-          lineNumber: LINE_NUMBER,
-          code: SOURCE_CODE
-        }
-      });
-    })();
-  `);
-
-  const extendDeclarationWithLocation = template(`
-    Object.defineProperty(TARGET, SYMBOL_NAME, {
-      value: {
-        fileName: FILENAME,
-        lineNumber: LINE_NUMBER
-      }
-    });
-  `)
+  /**
+   *  Genetares location descriptor
+   *
+   *  Object.defineProperty(TARGET, SYMBOL_NAME, {
+   *    value: {
+   *      fileName: FILENAME,
+   *      lineNumber: LINE_NUMBER,
+   *      sourceCode?: SOURCE_CODE
+   *    }
+   *  });
+   */
+  function createLocationExtender(node, useSymbol, fileName, lineNumber, sourceCode){
+    return t.callExpression(
```

Comment:
unfortunately, using template mechanism leads to cryptic error if we process yieldExpression and use `regenerator` package at the same time
`babel Expected type "Expression" with option {}`

Reply:
Could u explain how I could reproduce it so I could debug later? Exact input and options used (I guess we are talking about most recent version of this plugin). I have both today's and tomorrow's evenings already planned but I can work on it on Saturday and hopefully release it then

Reply:
just redefine `createLocationExtender` body to
```js
  function createLocationExtender(node, useSymbol, fileName, lineNumber, sourceCode){
    const extendExpressionWithLocation = template(`
      Object.defineProperty(TARGET, SYMBOL_NAME, {
          value: {
            fileName: FILENAME,
            lineNumber: LINE_NUMBER
          }
        });
    `);
    const extendedDeclaration = extendExpressionWithLocation({
        TARGET: node,
        SYMBOL_NAME: getSymbol(useSymbol),
        FILENAME: t.stringLiteral(fileName),
        LINE_NUMBER: t.numericLiteral(lineNumber),
      })

    return extendedDeclaration;
}
```
and run any existing test containing 
```js
    options: { presets: ['env'] },
```

Reply:
You should unpack `expression` from returned value, so:
```js
return extendExpressionWithLocation.expression;
```

Babel don't know how did you want to parse this code and it assumes statement here - in babel@7 there are actually various versions of `template`, ie. `template.expression`, `template.statement` and others - but we cant use them if we want to keep babel@6 compat for now.

Also It would be better for perf to cache `extendExpressionWithLocationTemplate` in outer scope - just before `createLocationExtender` declaration.

Reply:
I've pushed out the fix for this

Reply:
great explanation, I wish I could find that information before 👍